portality/lib/dataobj.py

```diff
@@ -959,7 +970,7 @@ def construct_validate(struct, context=""):
     }
     """
     # check that only the allowed keys are present
-    keys = struct.keys()
+    keys = list(struct.keys())
```

Comment:
probably not necessary to cat to list() since we're iterating through them next, but hopefully these will be caught as conflicts on the merge. It'll still work like this, too.